src/libraries/CurrencyDelta.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+
+library CurrencyDelta {
+    uint256 constant CURRENCY_DELTA_SLOT = uint256(keccak256("CurrencyDelta")) - 1;
+
+    function _computeSlot(address locker, Currency currency) internal pure returns (bytes32 hashSlot) {
+        uint256 slot = CURRENCY_DELTA_SLOT;
+
+        assembly {
```

Comment:
why is this done in assembly? cheaper?

Reply:
Yeeah its quite a lot cheaper. And its like another 1000 cheaper if instead we do 
```
        assembly {
            mstore(0, locker)
            mstore(32, currency)
            hashSlot := keccak256(0, 64)
        }
```

with a single hash and not have a `CURRENCY_DELTA_SLOT` thats added into the process with a second hash. But I feel like its definitely best practice to add that just for if we ever had another mapping with these 2 parameters too. Thoughts?

Reply:
This though is not standard for computing the mapping slot?

Reply:
Yeah the standard way of computing a mapping is including a slot